test/test-mc-range-mincover.c

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright 2022-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "test-mongocrypt.h"
+
+#include "mc-check-conversions-private.h"
+#include "mc-optional-private.h"
+#include "mc-range-mincover-private.h"
+#include "mc-array-private.h"
+
+/* Enable -Wconversion as error for only this file.
+ * Other libmongocrypt files warn for -Wconversion. */
+MC_BEGIN_CONVERSION_ERRORS
+
+typedef struct {
+   int32_t range_min;
+   int32_t range_max;
+   mc_optional_int32_t min;
+   mc_optional_int32_t max;
+   size_t sparsity;
+   /* expectMincoverString is newline delimitted list of strings. */
+   const char *expectMincoverString;
+   mc_array_t expectMincover;
+   const char *expectError;
+} Int32Test;
+
+static void
+Int32Test_dump (Int32Test *test, mc_mincover_t *got)
+{
+   fflush (stdout); // Avoid incomplete stdout output from prior tests on error
+   fprintf (stderr,
+            "testcase: range_min=%" PRId32 " range_max=%" PRId32,
+            test->range_min,
+            test->range_max);
+   if (test->min.set) {
+      fprintf (stderr, " min=%" PRId32, test->min.value);
+   }
+   if (test->max.set) {
+      fprintf (stderr, " max=%" PRId32, test->max.value);
+   }
+   fprintf (stderr, " sparsity=%zu", test->sparsity);
+   fprintf (stderr, " expected mincover:\n%s", test->expectMincoverString);
```

Comment:
Expected `mincover` does not have begin/end guards, whereas actual `mincover` has begin/end guards. Suggest making these consistent.

Reply:
Done.